### PR TITLE
Add explicit null returns in flutter/test

### PR DESCRIPTION
**Pull request by srawlins**
Adds null statements at the end of functions which have implicit null returns. These functions all have nullable return types (and not void, dynamic`, and upcoming analysis encourages such functions to have explicit returns).

*List which issues are fixed by this PR. You must list at least one issue.* https://github.com/flutter/flutter/issues/94759

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by srawlins** [comment]
CC @goderbauer

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
@gspencergoog Can you do a secondary review on this one?

File: packages/flutter/test/rendering/mouse_tracker_cursor_test.dart [hunk 1]
```diff
@@ -57,8 +57,11 @@ void main() {
   setUp(() {
     _binding.postFrameCallbacks.clear();
     _binding.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.mouseCursor, (MethodCall call) async {
-      if (_methodCallHandler != null)
+      if (_methodCallHandler != null) {
         return _methodCallHandler!(call);
+      } else {
+        return null;
+      }
```
Comment (gspencergoog) [inline packages/flutter/test/rendering/mouse_tracker_cursor_test.dart:64]:
```suggestion
      return null;
```

File: packages/flutter/test/widgets/modal_barrier_test.dart [hunk 2]
```diff
@@ -160,8 +160,11 @@ void main() {
     final List<String> playedSystemSounds = <String>[];
     try {
       tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
-        if (methodCall.method == 'SystemSound.play')
+        if (methodCall.method == 'SystemSound.play') {
           playedSystemSounds.add(methodCall.arguments as String);
+	} else {
+          return null;
+	}
```
Comment (gspencergoog) [inline packages/flutter/test/widgets/modal_barrier_test.dart:167]:
```suggestion
	}
        return null;
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Modulo the extra else clauses.

**Pull request merged**